kpatch-build/create-klp-module.c

```diff
@@ -491,17 +491,32 @@ int main(int argc, char *argv[])
 		remove_arch_sections(kelf);
 	}
 
+	/*
+	 * Group sections will need to update their sh_info fields with
+	 * new symbol table indexes, so stash their 'identifying entry'
+	 * symbol pointer for use when rebuilding the section.
+	 */
+	list_for_each_entry(sec, &kelf->sections, list) {
+		if (is_group_section(sec)) {
+			sec->sh_link_info = find_symbol_by_index(&kelf->symbols, sec->sh.sh_info);
+		}
+	}
+
 	remove_intermediate_sections(kelf);
```

Comment:
`kpatch_reindex_elements()` does a reindex of the symbols, after removing some sections from ELF, will it be better to move the loop to update `sec->sh_link_info` after `kpatch_reindex_elements()`

Reply:
ah !! right. I can not think of better name, probably drop prefix `sh_` from `sh_link_info`, as `sh` is redundant for section struct member.